### PR TITLE
perf: use mmkv for new data

### PR DESCRIPTION
**Pull request by skylarbarrera**
## What changed (plus any additional context for devs)
mmkv is orders of magnitude faster than AsyncStorage (our current local storage library)
source: https://github.com/mrousavy/StorageBenchmark

this is a slow migration to move new writes and reads over to mmkv without us having to do a larger migration (for dynamic keys) on app start up 

## Screen recordings / screenshots
https://cloud.skylarbarrera.com/Screen-Recording-2023-03-01-16-30-40.mp4

## What to test
moving from an old build to this branch should not cause any data interruptions and should not be noticeable from a user POV

### CODE REVIEW

**Comment by jkadamczyk** [comment]
So if I understand correctly, we have a bunch of leftovers in AsyncStorage due to some old legacy code, and we want to lazily migrate that over to MMKV. We will  eventually remove all this code at some point in time, since we want to use the new wrapper instead?

File: src/handlers/localstorage/common.ts [hunk 1]
```diff
@@ -46,14 +116,13 @@ export const getLocal = async (key = '', version = defaultVersion) => {
     });
     if (result && result.storageVersion === version) {
       return result;
-    }
-    if (result) {
-      removeLocal(key);
+    } else if (result && result.storageVersion !== version) {
```
Comment (jinchung) [inline src/handlers/localstorage/common.ts:119]: wouldn't need the additional `&& result.storageVersion !== version` check here given the result would've returned above on line 118

File: src/handlers/localstorage/common.ts [hunk 2]
```diff
@@ -62,13 +131,59 @@ export const getLocal = async (key = '', version = defaultVersion) => {
...
+ */
+export const getLocal = async (key = '', version = defaultVersion) => {
+  try {
+    const mmkvData = await getLocalMMKV({ key, version });
+
+    // if we have data in mmkv we want to use that
+    if (mmkvData) {
+      return mmkvData;
+    } else {
+      // otherwise we want to get the data from the legacy async storage
```
Comment (jinchung) [inline src/handlers/localstorage/common.ts:144]: placeholder note: did we want to add it to the local MMKV and then remove the local async here

**Comment by jkadamczyk** [comment]
I did test on Pixel 6 and it seems to work allright and the app feels pretty speedy in general. Switching wallets between mikearndt.eth and vitalik.eth is less than a second. Seems like everything is pretty neatly cached and read really fast

**Review by jkadamczyk — APPROVED** [review]

**Comment by jkadamczyk** [comment]
also works on iOS where I did have the app installed before

**Review by estrattonbailey — APPROVED** [review]

**Review by estrattonbailey — APPROVED** [review]

**Pull request merged**